core/src/main/java/jenkins/util/AntClassLoader.java

```diff
@@ -921,15 +919,14 @@ public URL getResource(String name) {
      * @return Url to found object
      */
     @CheckForNull
-    protected URL getUrl(Vector pathComponents, String name) {
-        Enumeration e = pathComponents.elements();
+    protected URL getUrl(ArrayList<File> pathComponents, String name) {
         URL url = null;
-        while (e.hasMoreElements() && url == null) {
-            File pathComponent = (File) e.nextElement();
+        for (File pathComponent : pathComponents) {
             url = getResourceURL(pathComponent, name);
-            if (url != null) {
-                log("Resource " + name + " loaded from ant loader", Project.MSG_DEBUG);
+            if (url == null) {
+                break;
```

Comment:
don't you want `continue` here? not really sure why this was inverted?

Reply:
Trying to deal w/ negations/loop control is hard.

What I was supposed to do was add the break and not change the condition.

I think I changed my plans for how to manage this loop repeatedly. Sadly, it appears that the version I put up for review was incomplete and incorrect.

The old code had a loop condition where once it hit `url == null`, would break out of the loop, this is what we want.

Reply:
Ah I see, missed that